src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryText.cs

```diff
@@ -487,50 +501,54 @@ public static bool TryText(FormattingInfo formatInfo, IRContext irContext, Formu
             return result != null;
         }
 
-        internal static bool TryExpandDateTimeExcelFormatSpecifiersToStringValue(IRContext irContext, string format, string defaultFormat, DateTime dateTime, TimeZoneInfo timeZoneInfo, CultureInfo culture, CancellationToken cancellationToken, out StringValue result)
+        internal static bool TryExpandDateTimeExcelFormatSpecifiersToStringValue(IRContext irContext, string format, string defaultFormat, DateTime dateTime, TimeZoneInfo timeZoneInfo, CultureInfo culture, CancellationToken cancellationToken, bool hasEnum, out StringValue result)
         {
             cancellationToken.ThrowIfCancellationRequested();
 
             result = null;
+
             if (format == null)
             {
                 result = new StringValue(irContext, dateTime.ToString(defaultFormat, culture));
                 return true;
             }
 
-            // DateTime format
-            switch (format.ToLowerInvariant())
+            // DateTime format has enum
+            if (hasEnum)
             {
-                case "'shortdatetime24'":
-                case "'shortdatetime'":
-                case "'shorttime24'":
-                case "'shorttime'":
-                case "'shortdate'":
-                case "'longdatetime24'":
-                case "'longdatetime'":
-                case "'longtime24'":
-                case "'longtime'":
-                case "'longdate'":
-                    var formatStr = ExpandDateTimeFormatSpecifiers(format, culture);
-                    result = new StringValue(irContext, dateTime.ToString(formatStr, culture));
-                    break;
-                case "'utc'":
-                case "utc":
-                    var formatUtcStr = ExpandDateTimeFormatSpecifiers(format, culture);
-                    result = new StringValue(irContext, ConvertToUTC(dateTime, timeZoneInfo).ToString(formatUtcStr, culture));
-                    break;
-                default:
-                    try
-                    {
-                        var stringResult = ResolveDateTimeFormatAmbiguities(format, dateTime, culture, cancellationToken);
-                        result = new StringValue(irContext, stringResult);
-                    }
-                    catch (FormatException)
-                    {
-                        return false;
-                    }
-
-                    break;
+                switch (format.ToLowerInvariant())
+                {
+                    case "'shortdatetime24'":

```

Comment:
>case [](http://example.com/codeflow?start=20&length=4)

nit-  (perhaps in future PR) - these are "magical" constants. It would be nice if we can share them across the source. 

Ie, instead of encoding as ("hasEnum" bool + string version of the enum), can we actually have a resolved enum and avoid the sttring literal compare?

Reply:
As discussed offline. I will leave this extra hasEnum here and will fix the enum type of datetimeformat enum for all source and refer then break this function to handle enum separately.